rust/private/rust.bzl

```diff
@@ -542,6 +542,20 @@ _common_attrs = {
         """),
         allow_files = True,
     ),
+    "dep_env_file": attr.label(
```

Comment:
I'm not sure this should be a common attr - if I'm understanding correctly, this is something a library sets to propagate to its dependents? It feels like it should either only live on `rust_library` (I don't see a use-case for `rust_binary` or `rust_test` setting a `dep_env_file`), or possibly be a standalone rule:

```starlark
cargo_dep_env(
    name = "set_b",
    dep_env_file = "set_b.env",
)
```

I'd lean towards the standalone rule, which just populates a `BuildInfo` provider, as this doesn't seem like it would be particularly coupled to a `rust_library`, but I could be easily convinced otherwise.